### PR TITLE
add a CLIENT_ONLY build option and handle for installer

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: cmake/macros/GenerateInstallers.cmake [hunk 1]
```diff
@@ -23,7 +23,7 @@ macro(GENERATE_INSTALLERS)
   set(CPACK_NSIS_DISPLAY_NAME ${_DISPLAY_NAME})
   set(CPACK_NSIS_PACKAGE_NAME ${_DISPLAY_NAME})
   if (PR_BUILD)
-    set(CPACK_NSIS_COMPRESSOR "/SOLID bzip2")
+    set(CPACK_NSIS_COMPRESSOR "bzip2")
```
Comment (birarda) [inline cmake/macros/GenerateInstallers.cmake]: The additional compression here isn't worth the extra compression (on build box) and decompression (on our machines) time

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8042/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8043/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8043/HighFidelity-Beta-PR12647-637c0f28fb68d7a9046dd16d437d941bff05e13c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8046/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8046/HighFidelity-Beta-PR12647-70455ac31ff1f33b0a44b9af6b94b657121725f9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8046/HighFidelity-Beta-PR12647-70455ac31ff1f33b0a44b9af6b94b657121725f9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8049/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8049/HighFidelity-Beta-PR12647-207833e69acb4a56bc3cdcdd34272ee2d175defd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8049/HighFidelity-Beta-PR12647-207833e69acb4a56bc3cdcdd34272ee2d175defd.exe

**Comment by birarda** [comment]
#### Test Plan

1. Download the PR build linked by Jenkins. 
2. Run the express install. Confirm that the express install launches the Interface and Sandbox and creates desktop shortcuts for the Interface and Sandbox. 
3. Confirm that the Interface and Sandbox run without crashing.
4. Run the uninstaller.
5. Run the custom install. Confirm that the "Choose Components" page shows both the Sandbox and the Interface.
6. Test each of the options on the custom install page on and off to confirm that they behave as you expect. You will need to run the custom installer multiple times, uninstalling between each run so that options between installs are not conflated.
7. Uninstall the custom install.
8. Download this client only installer from this PR (http://hifi-public.s3.amazonaws.com/birarda/HighFidelity-Beta-PR12647-client.exe).
9. Run the express install. Confirm that you get a desktop shortcut for Interface and that Interface launches when you complete the install.
10. Confirm that Interface runs without crashing on launch.
11. Confirm (by looking at the folder in Program Files) that there is no "assignment-client.exe" or "domain-server.exe" installed.

**Comment by qlimnavarro** [comment]
Currently testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
The installer from Step 8, after clicking express install I am presented with a blank screen with only the option to close. Custom install with the installer from test plan works as explained in the steps. 
capture

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8087/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8087/HighFidelity-Beta-PR12647-fc7daf59c1eb164f41e5a13a0bdaabe31097422d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8087/HighFidelity-Beta-PR12647-fc7daf59c1eb164f41e5a13a0bdaabe31097422d.exe

**Comment by birarda** [comment]
@qlimnavarro great catch - I've fixed it an re-linked a new client installer in the test plan. Could you please re-do the test once the newest build posts? We should re-do the regular installer tests as well.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8109/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8109/HighFidelity-Beta-PR12647-99345333eda1722b532042e5b1a7a4cdca7ef404.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8109/HighFidelity-Beta-PR12647-99345333eda1722b532042e5b1a7a4cdca7ef404.exe

**Comment by qlimnavarro** [comment]
Currently testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
Installing via the express installation in the client linked in Step 8, does not automatically launch the interface after install. Installing through custom and selecting "Launch interface after install" does launch the interface as expected. No logs were produced, if there is any other documentation that would help I'd be happy to provide.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8142/app_12647_PR.apk

**Comment by birarda** [comment]
Pushed a fix and linked a new version of the client installer (-v3)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8143/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8143/HighFidelity-Beta-PR12647-3d7628ffbfe6fe83401b5384ac919c07ae31cf60.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8143/HighFidelity-Beta-PR12647-3d7628ffbfe6fe83401b5384ac919c07ae31cf60.exe

**Comment by birarda** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8147/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8147/HighFidelity-Beta-PR12647-3d7628ffbfe6fe83401b5384ac919c07ae31cf60.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8147/HighFidelity-Beta-PR12647-3d7628ffbfe6fe83401b5384ac919c07ae31cf60.dmg

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
All tests pass and client now works as expected.

**Comment by birarda** [comment]
I rebased this because of other installer changes that came in, a re-test should be performed. I will post a new client installer as well (v4).

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8167/app_12647_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8167/HighFidelity-Beta-PR12647-cfdf8c25882fa5b62e5a7e8bd679f3fa9ec6a361.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8167/HighFidelity-Beta-PR12647-cfdf8c25882fa5b62e5a7e8bd679f3fa9ec6a361.exe

**Comment by birarda** [comment]
The new client link has been added to step 8 so that this is ready for a re-test.

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Client from build and from test plan passes tests as expected

**Pull request merged**